docs/ARCHITECTURE.md

```diff
@@ -50,6 +50,27 @@ service manager.
 
 BPF helpers written in C and used by PID 1 can be found under `src/core/bpf/`.
 
+### Implementing Directives
+
+The system and session manager supports a large number of directives. These can generally
+be configured in three ways:
+
+1) Via textual, INI-style configuration files called `units`
+2) Via DBUS messages to the manager
+3) Via the `systemd-run` and `systemctl set-property` commands
```

Comment:
this is the same as 2 of course, since those tools just wrap D-Bus. I think it deserves mentioning, since from a user perspective these are two things, though internally they reach PID 1 the same way.

Reply:
I'll mention it, but this is aimed at developers, to give a quick hint on how to get started when one wants to add a new setting. From what I hear from others, and my own experience in years past, the way one does that right now is to pick a commit that added a new settings and checking what files it changed, which is not the most efficient!